source/tools/toolutil/meson.build

```diff
@@ -0,0 +1,43 @@
+sources = files(
+  'filestrm.cpp',
+  'package.cpp',
+  'pkgitems.cpp',
+  'swapimpl.cpp',
+  'toolutil.cpp',
+  'unewdata.cpp',
+  'collationinfo.cpp',
+  'denseranges.cpp',
+  'ucm.cpp',
+  'ucmstate.cpp',
+  'uoptions.cpp',
+  'uparse.cpp',
+  'ucbuf.cpp',
+  'xmlparser.cpp',
+  'writesrc.cpp',
+  'pkg_icu.cpp',
+  'pkg_genc.cpp',
+  'pkg_gencmn.cpp',
+  'ppucd.cpp',
+  'flagparser.cpp',
+  'filetools.cpp',
+  'udbgutil.cpp',
+  'dbgutil.cpp',
+  'ucln_tu.cpp',
+)
+
+toolutil_c_args = ['-DU_TOOLUTIL_IMPLEMENTATION']
+toolutil_cpp_args = ['-DU_TOOLUTIL_IMPLEMENTATION']
```

Comment:
Hmm, any particular reason to define this twice? Looks like upstream rigidly defines all these *_IMPLEMENTATION defines regardless of language used, it's more like a toolutil_component_args.

Reply:
That's just how it was done generally. I think this was just preparing for the case where C and C++ might need different compiler flags. Dunno if that's going to come up.